source/v5/mqtt5_to_mqtt3_adapter.c

```diff
@@ -2226,23 +2234,22 @@ static void s_aws_mqtt5_to_mqtt3_adapter_subscribe_completion_fn(
             "id=%p: mqtt3-to-5-adapter, completing multi-topic subscribe",
             (void *)adapter);
 
+        AWS_VARIABLE_LENGTH_ARRAY(struct aws_mqtt_topic_subscription, multi_sub_subscription_buf, reason_code_count);
```

Comment:
With `reason_code_count` equal to 0, the expression here will be
```c
struct aws_mqtt_topic_subscription multi_sub_subscription_buf[0];
```
, which is invalid.

It's just another reason to handle `reason_code_count == 0` separately.

Reply:
can't we have a reason code of zero without suback being null?

Reply:
According to [the docs](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901178), the number of reason codes must match the number of topics. Though, I'd say a simple check probably won't hurt.